### ISSUE TITLE
[4pt] Parameterized workflow for deploying to test

### ISSUE THREAD

**Issue by NickVolynkin** [tarantool/doc#2155]
# What needs to be done

Parameterized workflow for deploying to test. Accepts git_ref value to checkout in each submodule and the environment name.

# Definition of done

- [x] rewrite pull_submodules.sh in Python
- [x] pull_submodules.py accept parameters as environment variables
- [x] pull_submodules.py should print stdout and stderr from spawned processes, and it should exit on error
- [x] add new workflow so that it accepts environment name and git_ref's and sets environment variables from them
- [ ] use the new workflow to deploy branches from tarantool/doc repository (by running it from deploy-branch workflow with required parameters)
- [x] the `deploy` and `deploy-branches` workflows can use the new `pull_modules.py` 

> add new workflow so that it accepts environment name and git_ref's and sets environment variables from them

The idea here is that this function-workflow accepts a set of GIT_REF's and also the environment name as parameters. After that, we **can** rewrite `deploy-branch` in tarantool/doc so that it calls the function-workflow, giving it the environment name and running it on the same branch. But we can do it later.

The DoD of this issue is that there's a **new** workflow which we can run at least manually. Then we can go on with calling it from workflows in other repos.

**Comment by artembo** [comment]
Workflow dispatch inputs defined according to https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs

**Comment by NickVolynkin** [comment]
Logs from `pull_modules.py` can be improved.

```
Scanning dependencies of target pull-modules
[100%] Update submodules to their latest commits
+ git checkout origin/master
Warning: you are leaving 1 commit behind, not connected to
any of your branches:

  b8bc178 Fix documentation build warnings (#1401)

If you want to keep it by creating a new branch, this may be a good time
to do so with:

 git branch <new-branch-name> b8bc178

HEAD is now at 0fc7cbc Stop roles with `on_shutdown` trigger (#1428)
+ git checkout origin/master
Previous HEAD position was 5770f5e Release 2.9.1 (#548)
HEAD is now at 1b67af5 Fix broken link (build Tarantool from source) in README (#550)
+ git checkout origin/master
HEAD is now at 983bdf6 Merge pull request #59 from tarantool/43-prometheus-alert-rules
+ git checkout origin/master
Warning: you are leaving 1 commit behind, not connected to
any of your branches:

  5b9b94c Write execution time for every test (#130)

If you want to keep it by creating a new branch, this may be a good time
to do so with:

 git branch <new-branch-name> 5b9b94c

HEAD is now at 63f50c5 Repeat tests' names (#133)
+ git checkout origin/master
Warning: you are leaving 1 commit behind, not connected to
any of your branches:

  c00e60d cpu metrics: fix hot reload (#237)

If you want to keep it by creating a new branch, this may be a good time
to do so with:

 git branch <new-branch-name> c00e60d

HEAD is now at f57e7ea fiber.yield() call removed from quantile (#241)
+ git checkout origin/master
HEAD is now at 3d058b1 doc: fix build warnings (#101)
+ git checkout origin/master
HEAD is now at 4d3365e doc: fix documentation build warnings (#12)
[100%] Built target pull-modules
```

**Comment by artembo** [comment]
Fixed in 7de13e68ff126079e97cb22ac78cb855c38b6751
